library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/Section.java

```diff
@@ -316,6 +334,46 @@ public final int getSectionItemsTotal() {
      */
     public abstract int getContentItemsTotal();
 
+    /**
+     * Creates the View for a single Item. This must be implemented if and only if
+     * {@link #isItemViewWillBeProvided()} is true.
+     *
+     * @param parent The parent view. Note that there is no need to attach the new view.
+     * @return View for an Item of this Section.
+     */
+    public View getItemView(ViewGroup parent) {
+        throw new UnsupportedOperationException(
```

Comment:
could you please add unit tests for these getter methods

Reply:
That's a practically empty method, not sure why I should add a unit test.

If you mean a test that checks that they're called (i.e. testing the behavior of SectionedRecyclerViewAdapter), I could do that. The spy/stub patterns you have entails plenty of boilerplate. Have you considered Mockito?

Reply:
There are good points, I agree that these tests are unnecessary and that these onBind* tests should be moved to the adapter class. I'm also ok to have all the spy/stubs replaced by mockito.

Reply:
OTOH SectionedRecyclerViewAdapterTest is already huge. I'll leave them where they are but try and convert to Mockito.

Reply:
PTAL. I think the Mockito code is nicer to read, plus it allows for more fine-grained assertions. In fact, Mockito was already a gradle dependency. If you're good with these changes, I'll build the new tests with Mockito also.

The StatelessSectionTest also contains analogous tests for its onBind*ViewHolder() methods. I'm not sure why these methods are explicitly overridden in the StatelessSection and why they need separate testing. Maybe it's for historical reasons? It looks like everything except the constructor could be removed in StatelessSection, and the respective *StatelessSectionSpy classes and unit tests could also be removed. WDYT?

Reply:
Legend! 🏅 

- Tests with Mockito: yes, please go ahead.

- StatelessSectionTests: seem to be redundant and can be removed.

- Overriden methods in StatelessSection: they were overridden with the final keyword to avoid users to try to override them, should have added a comment there.

Reply:
We could potentially kill StatelessSection and add a flag in Section in order to know if it has state or not.
Then move the validations of StatelessSection constructor to SectionParameters.Builder. And  add a "withState" method in SectionParameters.Builder.

It won't be that disruptive as users would only have to change their classes to extend Section instead of StalessSection and add call the withState method from the builder.

But definitely work for a separate PR.

Reply:
PTAL. I only added tests for getItemView(Holder) for now. Has full test coverage except the `if (resId == null)` branch, which cannot properly be mocked because `getItemResourceId` is final, but then it's already guarded against by SectionParameters.

Should I duplicate (some of) these tests for the other view kinds (header/footer/...)?

Reply:
Good stuff, yes, please!

Reply:
Ok, added those as well, with a few variations.